baseplate/experiments/providers/__init__.py

```diff
@@ -55,28 +67,30 @@ def parse_experiment(config):
     :return: A subclass of :py:class:`Experiment` for the given experiment
         type.
     """
-    experiment_type = config["type"].lower()
-    experiment_id = config["id"]
+    experiment_type = config.get("type")
+    if experiment_type:
+        experiment_type = experiment_type.lower()
+    experiment_id = config.get("id")
     assert isinstance(experiment_id, int)
-    name = config["name"]
+    name = config.get("name")
     owner = config.get("owner")
-    if "start_ts" in config and "stop_ts" in config:
-        start_ts = config["start_ts"]
-        stop_ts = config["stop_ts"]
-    elif "expires" in config:
-        warn_deprecated(
-            "The 'expires' field is in experiment %s deprecated, you should "
-            "use 'start_ts' and 'stop_ts'." % name
-        )
-        start_ts = time.time()
-        expires = datetime.strptime(config["expires"], ISO_DATE_FMT)
-        epoch = datetime(1970, 1, 1)
-        stop_ts = (expires - epoch).total_seconds()
-    else:
-        raise ValueError(
-            "Invalid config for experiment %s, missing start_ts and/or "
-            "stop_ts." % name
-        )
+    start_ts = config.get("start_ts")
+    stop_ts = config.get("stop_ts")
+    if (start_ts is None or stop_ts is None):
```

Comment:
nit: no parens around the if condition in python please!

Reply:
fixed. This used to be multi-line.

Reply:
this doesn't appear fixed in the current diff?